File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/ShTest.java
package com.panopset.tests.blackjackEngine;

import static com.panopset.blackjackEngine.CommandDefinitions.CMD_DEAL;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_HIT;
import static com.panopset.blackjackEngine.CommandDefinitions.CMD_STAND;
import org.junit.jupiter.api.Test;

import com.panopset.blackjackEngine.BlackjackConfigDefault;
import com.panopset.blackjackEngine.BlackjackGameEngine;

public class ShTest extends AbstractBlackjackTest {

  @Test
  void test() {
    BlackjackGameEngine bge = new BlackjackGameEngine(new BlackjackConfigDefault());
    verifyRecommendedActions(bge, new String[] {CMD_DEAL, CMD_HIT, CMD_STAND},
        stackHard16_with_4_vT());
    verifyRecommendedActions(bge, new String[] {CMD_DEAL, CMD_HIT, CMD_STAND},
        stackHard16_with_5_vT());
    verifyRecommendedActions(bge, new String[] {CMD_DEAL, CMD_HIT}, stackHard16vT());
  }

}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/ShTest.kt
package com.panopset.tests.blackjackEngine

import com.panopset.blackjackEngine.*
import org.junit.jupiter.api.Test

class ShTest {
    @Test
    fun test() {
        val bge = BlackjackGameEngine(BlackjackConfigDefault())
        verifyRecommendedActions(
            bge, arrayOf(CMD_DEAL, CMD_HIT, CMD_STAND),
            stackHard16_with_4_vT()
        )
        verifyRecommendedActions(
            bge, arrayOf(CMD_DEAL, CMD_HIT, CMD_STAND),
            stackHard16_with_5_vT()
        )
        verifyRecommendedActions(bge, arrayOf(CMD_DEAL, CMD_HIT), stackHard16vT())
    }
}
